_compdemos/rslurm_example.md

```diff
@@ -0,0 +1,200 @@
+                                 col_names=samp)) #set column name ar
+
+  write.table(counts, file=paste0(samp,"_abundance.tsv"), quote = FALSE, sep = "\t")
+}
+```
+
+
+## Rslurm Set-up
+
+```
+#install the package if necessary and load it into the R session
+# install.packages('rslurm')
+library(rslurm)
+```
+
+To do this, you will need specify some parameters for SLURM job scheduler. I usually get these from my batch scripts header lines that begin with `#SBATCH`. More details and examples of a batch script can be found [here](https://github.com/FredHutch/slurm-examples/tree/master/R_and_sh_example).
+
+I add a lot of options since I tend to like to modify these parameters fairly often, but it is completely not necessary to go into this level of detail. You can simply request a number of node(s) and time limit and you will be all set.
+
+```
+#Create a list of SBATCH options
+sopt <- list('nodes'='1',  #select 1 node
+             'cpus-per-task'='1', #select 1 CPU for the job
+             'partition'='campus', #campus gizmo nodes
+             'mem'='6G', #memory request of 6Gb
+             'time' = '24:00:00', #24 hours time limit chosen.
+             'mail-type'='END,FAIL', #only email me if the job completes or fails for any reason
+             'mail-user'='USER@fredhutch.org') #input your email here.
+
+```
+
+Now you can set up your Rslurm example. You just need to have the appropriate information ready to feed into `tximport`. Of note, `Rslurm` creates a directory in your current working directory with LOTS of files.  You may want to check `getwd()` to ensure that you are not creating huge files in a directory that really cannot support it. This becomes and issue when you are doing this with say 1000's of samples or doing a different analysis that just outputs a ton of large files.
+
+Use `list.files()` to create a list of the abundance text files. I used  `recursive=FALSE` here, but you may need to change that since you likely will have data in nested directories. Also, adding sample names here will be used for the column names in the resulting output.
+
+```
+#list full file paths for each counts file
+files <- list.files(path = getwd(),
+             pattern = "abundance.tsv$",
+             recursive = FALSE,
+             full.names = TRUE)
+
+#Add names to be the sample names
+names(files) <- gsub("^.+\\/(sample[0-9]{1,2})_abund.+","\\1", files)
+
+
+files[1:2]
+```
+
+Rslurm will create a directory containg all the accessory data files, libraries to load, and the batch script for you. I have `submit=TRUE` assuming you are running this on a Rhino/GIZMO node.
+
+## Submit Job to Scheduler
+
+```
+txi.geneLevel.job <- slurm_call(f=tximport, #function to be used
+                     jobname = "tximport_Gene", #name of the  output directory
+
+                    #Define parameters for the function here, based on the exact argument names, etc in the ?tximport documentation.
+                     params =  list(files = files,
```

Comment:
Question: Is the `params` argument the container for the input argument for `tximport`?

Reply:
Yup!  Discussed the other day.  Going to close this.